spec/migrations/20220520105024_update_service_user_roles_spec.rb

```diff
@@ -0,0 +1,48 @@
+require_migration
+
+describe UpdateServiceUserRoles do
+  migration_context :up do
+    let(:user_role_stub) { migration_stub(:MiqUserRole) }
+    let(:product_feature_stub) { migration_stub(:MiqProductFeature) }
+
+    let!(:other) { product_feature_stub.create!(:feature_type => "node", :identifier => "other") }
+    let!(:service) { product_feature_stub.create!(:feature_type => "node", :identifier => "service") }
+    let!(:service_accord) { product_feature_stub.create!(:feature_type => "node", :identifier => "service_accord") }
+
+    it "does nothing to roles without service_accord" do
+      user_role = user_role_stub.create!(:miq_product_features => [other], :read_only => false)
+
+      migrate
+      user_role.reload
+
+      expect(user_role.miq_product_features).to match_array([other])
+    end
+
+    it "converts service_accord to service" do
+      user_role = user_role_stub.create!(:miq_product_features => [service_accord], :read_only => false)
+
+      migrate
+      user_role.reload
+
+      expect(user_role.miq_product_features).to match_array([service])
+    end
+
+    it "converts service_accord to service without duplicate" do
+      user_role = user_role_stub.create!(:miq_product_features => [other, service, service_accord], :read_only => false)
+
+      migrate
+      user_role.reload
+
+      expect(user_role.miq_product_features).to match_array([other, service, service])
```

Comment:
We are testing the line I mentioned above. Specifically, if the user is already part of a service, then lets not add the service again.

```suggestion
      expect(user_role.miq_product_features).to match_array([other, service])
```

Reply:
since its expecting 3 items, can we use like this

```
expect(user_role.miq_product_features.uniq).to match_array([other, service])
expect(user_role.miq_product_features).not_to match_array([service_accord])
```